.travis.yml

```diff
@@ -0,0 +1,237 @@
+# granted,  to either  redistribute  or  modify this  file,  provided that  the
+# conditions  mentioned in  the file  COPYING are  met.  Consult  the  file for
+# details.
+
+# We only manage non-amd64 free pipelines.
+# https://docs.travis-ci.com/user/billing-overview/
+
+language: c
+
+os: linux
+
+dist: focal
+
+git:
+  quiet: true
+
+cache:
+  ccache: true
+  directories:
+    - $HOME/config_2nd
+    - $HOME/.downloaded-cache
+
+env:
+  global:
+    # Reset timestamps early
+    - _=$(touch NEWS && find . -type f -exec touch -r NEWS {} +)
+    - CONFIGURE_TTY=no
+    - CCACHE_COMPILERCHECK=none
+    - CCACHE_NOCOMPRESS=1
+    - CCACHE_MAXSIZE=512Mi
+    - NPROC="`nproc`"
+    # JOBS and SETARCH are overridden when necessary; see below.
+    - JOBS=-j$((1+${NPROC}))
+    - SETARCH=
+    - RUBY_PREFIX=/tmp/ruby-prefix
+    - GEMS_FOR_TEST='timezone tzinfo'
+    - UPDATE_UNICODE="UNICODE_FILES=. UNICODE_PROPERTY_FILES=. UNICODE_AUXILIARY_FILES=. UNICODE_EMOJI_FILES=."
+    # The tests skipped in `make test-all`.
+    - TEST_ALL_SKIPPED_TESTS=
+    # Run the separated tests with allowing failures.
+    - RUN_SEPARATED_TESTS=true
+    # The tests executed separately by `make test-all`.
+    - TEST_ALL_SEPARATED_TESTS=
+    # https://github.com/travis-ci/travis-build/blob/e411371dda21430a60f61b8f3f57943d2fe4d344/lib/travis/build/bash/travis_apt_get_options.bash#L7
+    - travis_apt_get_options='--allow-downgrades --allow-remove-essential --allow-change-held-packages'
+    - travis_apt_get_options="-yq --no-install-suggests --no-install-recommends $travis_apt_get_options"
+    # -O1 is faster than -O3 in our tests.
+    - optflags=-O1
+    # -g0 disables backtraces when SEGV.  Do not set that.
+    - debugflags=-ggdb3
+
+.org.ruby-lang.ci.matrix-definitions:
+
+  - &gcc-10
+    compiler: gcc-10
+    before_install:
+      - tool/travis_retry.sh sudo bash -c "rm -rf '${TRAVIS_ROOT}/var/lib/apt/lists/'* && exec apt-get update -yq"
+      - >-
+        tool/travis_retry.sh sudo -E apt-get $travis_apt_get_options install
+        ccache
+        gcc-10
+        g++-10
+        libffi-dev
+        libgdbm-dev
+        libncurses-dev
+        libncursesw5-dev
+        libreadline-dev
+        libssl-dev
+        libyaml-dev
+        openssl
+        zlib1g-dev
+
+  # --------
+
+  - &arm64-linux
+    name: arm64-linux
+    arch: arm64
+    <<: *gcc-10
+
+  - &ppc64le-linux
+    name: ppc64le-linux
+    arch: ppc64le
+    <<: *gcc-10
+    env:
+      # FIXME: Skip some failing TestGCCompact tests.
+      # https://bugs.ruby-lang.org/issues/17871
+      - TEST_ALL_SKIPPED_TESTS="test_gc_compact_stats test_complex_hash_keys test_ast_compacts test_compact_count"
+      # The tests crash the process.
+      - RUN_SEPARATED_TESTS=false
+
+  - &s390x-linux
+    name: s390x-linux
+    arch: s390x
+    <<: *gcc-10
+
+  - &arm32-linux
+    name: arm32-linux
+    arch: arm64
+    # https://packages.ubuntu.com/xenial/crossbuild-essential-armhf
```

Comment:
Self review: Replace the outdated URL with https://packages.ubuntu.com/focal/crossbuild-essential-armhf .